vllm/attention/backends/flash_attn.py

```diff
@@ -211,6 +211,12 @@ def decode_metadata(self) -> Optional["FlashAttentionMetadata"]:
             block_tables=self.block_tables[self.num_prefills:],
             use_cuda_graph=self.use_cuda_graph,
         )
+        # Batch may be composed of prefill|decodes, adjust query start indices
```

Comment:
nit- Can this be rewritten in L206 as? If possible same for the other backends.

``` 
query_start_loc=(self.query_start_loc[self.num_prefills:] - self.query_start_loc[self.num_prefills])
            if self.query_start_loc is not None else None,

```

Reply:
good one! I would leave the other backends as is, as `query_start_loc` is not yet enabled on rocm while xformers uses pagedattention for decodes.